src/main/java/org/apache/sling/installer/provider/jcr/impl/JcrInstaller.java

```diff
@@ -707,6 +716,7 @@ private UpdateResult handleUpdate(final String resourceType,
             dataNode.setProperty(PROP_MODIFIED, Calendar.getInstance());
             dataNode.setProperty(PROP_ENC, ENCODING);
             dataNode.setProperty(PROP_MIME, MIME_TXT);
+            dataNode.setProperty(ORIGINAL_PID, id);
```

Comment:
Does this work, seeting a property on the `jcr:content` node? I think the `nt:resource` node type is quite restrictive. IIRC Oak uses a different node-type for a file's `jcr:content` node. Maybe that one is more lenient. Alternatively, a mixin nodetype could be added to allow for the extra property.

Reply:
I didn't get any exceptions while creating a config on AEM 1 and it gets replicated on AEM 2 successfully.

Reply:
I suggest we drop the original.pid. Instead we go with your approach of checking for the legacy pid (https://github.com/apache/sling-org-apache-sling-installer-factory-configuration/pull/13/files#diff-ddf6029cf7d9aa5ed2720412a7fe8de77d7f5df784c32864c6abc54393e78321R95-R97).

That isn't quite as generic as I had originally intended, but it should be good enough. Unless I am missing something.